### PR TITLE
Added zocl aie error capturing and reporting

### PR DESCRIPTION
**Pull request by larry9523**
This PR 
1) Added register AIE error call back function
2) Cache and opulate new AIE class errors in the call back function when AIE asynchronous error happens
3) Change zocl async error recording to only record one last error for each class

### CODE REVIEW

**Review by stsoe — APPROVED** [review]

**Comment by larry9523** [comment]
retest this please

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 1]
```diff
@@ -99,6 +221,19 @@ zocl_create_aie(struct drm_zocl_dev *zdev, struct axlf *axlf)
 			DRM_ERROR("Fail to allocate memory.\n");
 			goto done;
 		}
+
+		zdev->aie->err.errors = vzalloc(sizeof (struct aie_error) *
+		    ZOCL_AIE_ERROR_CACHE_CAP);
```
Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:226]: Why the word wrap?  Would it be easier to read if the code as all on one line.

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 2]
```diff
@@ -131,9 +266,15 @@ zocl_create_aie(struct drm_zocl_dev *zdev, struct axlf *axlf)
 
 	zdev->aie->partition_id = req.partition_id;
 	zdev->aie->uid = req.uid;
+
+	/* Register AIE error call back function. */
+	rval = aie_register_error_notification(zdev->aie->aie_dev,
```
Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:271]: Why the word wrap? Would it be easier to read if the code as all on one line.

**Review by rozumx — APPROVED** [review]
No issues with the code as written.  Did encounter though 2 locations in the code base where the functionality was split into 2 lines.  Made it a little difficult to read.  Aside from that MINOR issue, all is good.

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c (hunk 1 above)
Comment (larry9523) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c:226]: Linux kernel code has a coding style that requires breaking long lines (maximum 80 characters per line). checkpatch tool will also enforce it.

https://www.kernel.org/doc/html/v4.10/process/coding-style.html

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c (hunk 2 above)
Comment (larry9523) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c:271]: same as above

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c (hunk 1 above)
Comment (rozumx) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c:226]: Thank you for the reminder.

**Review by mamin506 — APPROVED** [review]

**Pull request merged**